src/app/forms/components/defect-select/defect-select.component.ts

```diff
@@ -1,24 +1,57 @@
-import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
 import { Defect } from '@models/defects/defect.model';
 import { Deficiency } from '@models/defects/deficiency.model';
 import { Item } from '@models/defects/item.model';
+import { VehicleTypes } from '@models/vehicle-tech-record.model';
+import { Store } from '@ngrx/store';
+import { DefectsState, filteredDefects } from '@store/defects';
+import { selectedTestResultState } from '@store/test-records';
+import { TestResultsState } from '@store/test-records/reducers/test-records.reducer';
+import { takeUntil, filter, Subject } from 'rxjs';
 
 @Component({
-  selector: 'app-defect-select[isParentEditing][defects]',
+  selector: 'app-defect-select',
   templateUrl: './defect-select.component.html',
   styleUrls: ['./defect-select.component.scss']
 })
-export class DefectSelectComponent {
-  @Input() defects!: Defect[] | null;
-  @Input() isParentEditing = false;
-  @Output() formChange = new EventEmitter<{ defect: Defect, item: Item, deficiency?: Deficiency }>();
-
+export class DefectSelectComponent implements OnInit, OnDestroy {
+  defects: Defect[] = [];
   isEditing = false;
   selectedDefect?: Defect;
   selectedItem?: Item;
   selectedDeficiency?: Deficiency;
+  vehicleType!: VehicleTypes;
+
+  onDestroy$ = new Subject();
+
+  constructor(
+    private testResultsStore: Store<TestResultsState>,
+    private defectsStore: Store<DefectsState>,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit(): void {
+    this.testResultsStore
+      .select(selectedTestResultState)
+      .pipe(
+        takeUntil(this.onDestroy$),
+        filter(testResult => !!testResult)
+      )
+      .subscribe(testResult => {
+        this.vehicleType = testResult!.vehicleType;
+      });
+
+    this.defectsStore.select(filteredDefects(this.vehicleType)).subscribe(defectsTaxonomy => {
+      this.defects = defectsTaxonomy;
+    });
```

Comment:
```suggestion
    this.defectsStore
      .select(filteredDefects(this.vehicleType))
      .subscribe(defectsTaxonomy => this.defects = defectsTaxonomy);
```
To make it a bit more clearer.